test/wallet/signer.ts

```diff
@@ -0,0 +1,300 @@
+    // Test case data generated using this tutorial - https://xrpl.org/send-xrp.html#send-xrp
+    const tx3: Transaction = {
+      TransactionType: "Payment",
+      Account: "rHLEki8gPUMnF72JnuALvnAMRhRemzhRke",
+      Amount: "22000000",
+      Destination: "rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe",
+      Flags: 2147483648,
+      LastLedgerSequence: 20582339,
+      Fee: "12",
+      Sequence: 20582260,
+    };
+    const signedTxBlob =
+      "120000228000000024013A0F74201B013A0FC36140000000014FB18068400000000000000C732102A8A44DB3D4C73EEEE11DFE54D2029103B776AA8A8D293A91D645977C9DF5F544744730450221009ECB5324717E14DD6970126271F05BC2626D2A8FA9F3797555D417F8257C1E6002206BDD74A0F30425F2BA9DB69C90F21B3E27735C190FB4F3A640F066ACBBF06AD98114B3263BD0A9BF9DFDBBBBD07F536355FF477BF0E98314F667B0CA50CC7709A220B0561B85E53A48461FA8";
+
+    const wallet = Wallet.fromSeed(seed);
+
+    const signedTx: string = sign(wallet, tx3);
+
+    assert.equal(signedTx, signedTxBlob);
+  });
+
+  it("sign in multisign format", function () {
+    const wallet = Wallet.fromSeed(unsignedSecret1);
+
+    assert.deepEqual(
+      decode(sign(wallet, unsignedTx1, true)),
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Transaction is equivalent to JsonObject
+      multisignTx1 as unknown as JsonObject
+    );
+  });
+
+  it("multisign runs successfully with Transaction objects", function () {
+    const transactions: Transaction[] = [
+      multisignTxToCombine1,
+      multisignTxToCombine2,
+    ];
+
+    assert.deepEqual(multisign(transactions), expectedMultisign);
+  });
+
+  it("multisign runs successfully with tx_blobs", function () {
+    const transactions: Transaction[] = [
+      multisignTxToCombine1,
+      multisignTxToCombine2,
+    ];
+
+    const encodedTransactions: string[] = transactions.map(encode);
+
+    assert.deepEqual(multisign(encodedTransactions), expectedMultisign);
+  });
+
+  it("multisign throws a validation error when there are no transactions", function () {
+    const transactions: Transaction[] = [];
+    assert.throws(() => multisign(transactions), ValidationError);
+  });
+
+  it("multisign throws when trying to combine two different transactions", function () {
+    const differentMultisignedTx: Transaction = {
+      TransactionType: "Payment",
+      Sequence: 1,
+      Amount: "20000000",
+      Fee: "12",
+      SigningPubKey: "",
+      Account: "rhvh5SrgBL5V8oeV9EpDuVszeJSSCEkbPc",
+      Destination: "rQ3PTWGLCbPz8ZCicV5tCX3xuymojTng5r",
+      Signers: [
+        {
+          Signer: {
+            SigningPubKey:
+              "02A8A44DB3D4C73EEEE11DFE54D2029103B776AA8A8D293A91D645977C9DF5F544",
+            TxnSignature:
+              "3044022077BCE143B9A0B51A7716BB93CBC0C99FB41BA339D91A87CB9E47DA80A7EF660802205C81AA49D408771F65A131200CCBFC536ACFE212C1414E05E43B56BE1F9380F2",
+            Account: "rHLEki8gPUMnF72JnuALvnAMRhRemzhRke",
+          },
+        },
+      ],
+    };
```

Comment:
Given that we just discussed this, add a test to make sure that transactions signed in non-multisign mode are errored correctly (with the correct error message)